x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/latest_function_form.tsx

```diff
@@ -51,25 +57,51 @@ export const LatestFunctionForm: FC<LatestFunctionFormProps> = ({ latestFunction
             defaultMessage="Sort field"
           />
         }
+        helpText={i18n.translate('xpack.transform.stepDefineForm.sortHelpText', {
+          defaultMessage: 'Select the date field to be used to identify the latest document.',
+        })}
       >
-        <EuiComboBox
-          fullWidth
-          placeholder={i18n.translate('xpack.transform.stepDefineForm.sortPlaceholder', {
-            defaultMessage: 'Add a sort field ...',
-          })}
-          singleSelection={{ asPlainText: true }}
-          options={latestFunctionService.sortFieldOptions}
-          selectedOptions={
-            latestFunctionService.config.sort ? [latestFunctionService.config.sort] : []
-          }
-          onChange={(selected) => {
-            latestFunctionService.updateLatestFunctionConfig({
-              sort: { value: selected[0].value, label: selected[0].label as string },
-            });
-          }}
-          isClearable={false}
-          data-test-subj="transformWizardSortFieldSelector"
-        />
+        <>
+          {latestFunctionService.sortFieldOptions.length > 0 && (
+            <EuiComboBox
+              fullWidth
+              placeholder={i18n.translate('xpack.transform.stepDefineForm.sortPlaceholder', {
+                defaultMessage: 'Add a sort field ...',
+              })}
+              singleSelection={{ asPlainText: true }}
+              options={latestFunctionService.sortFieldOptions}
+              selectedOptions={
+                latestFunctionService.config.sort ? [latestFunctionService.config.sort] : []
+              }
+              onChange={(selected) => {
+                latestFunctionService.updateLatestFunctionConfig({
+                  sort: { value: selected[0].value, label: selected[0].label as string },
+                });
+              }}
+              isClearable={false}
+              data-test-subj="transformWizardSortFieldSelector"
+            />
+          )}
+          {latestFunctionService.sortFieldOptions.length === 0 && (
+            <EuiCallOut color="danger" iconType="alert" size="m">
+              <p>
+                <FormattedMessage
+                  id="xpack.transform.stepDefineForm.sortFieldOptionsEmptyError"
+                  defaultMessage="No date fields available to sort on. To override the configuration with another field type, please copy the configuration to the clipboard using the following button and continue creating the transform in Kibana Dev Console."
```

Comment:
I think this can be shortened. For example:
```suggestion
                  defaultMessage="No date fields are available to sort on. To use another field type, copy the configuration to the clipboard and continue creating the transform in the Console."
```